frontend/src/main/web/lib/stores/GlossaryStore.js

```diff
@@ -0,0 +1,522 @@
+  return saveOrUpdateGlossary(glossary);
+}
+
+function updateGlossary(entry) {
+  entry.status.isSaving = true;
+  var glossary = GlossaryHelper.generateGlossaryDTO(entry);
+  return saveOrUpdateGlossary(glossary);
+}
+
+function saveOrUpdateGlossary(glossary) {
+  var url = Configs.baseUrl + "/glossary/" + Configs.urlPostfix;
+  return new Promise(function(resolve, reject) {
+    Request.post(url)
+      .set('Content-Type', 'application/json')
+      .set('Accept', 'application/json')
+      .send(glossary)
+      .end(function (err, res) {
+        if(err != null && err.error === true) {
+          console.error(url, err);
+        }
+        if(res != null) {
+          if (res.error) {
+            console.error(url, res.status, res.error.toString());
+            reject(Error(res.error.toString()));
+          } else {
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+  });
+}
+
+function uploadFile(data) {
+  var url = Configs.baseUrl + "/glossary/upload",
+    uploadFile = data.uploadFile;
+
+  _state['uploadFile']['status'] = 0;
+
+  return new Promise(function(resolve, reject) {
+    Request.post(url)
+      .attach('file', uploadFile.file, uploadFile.file.name)
+      .field('fileName', uploadFile.file.name)
+      .field('srcLocale', data.srcLocale)
+      .field('transLocale', data.uploadFile.transLocale)
+      .set('Accept', 'application/json')
+      .end(function (err, res) {
+        _state['uploadFile']['status'] = -1;
+        _state['uploadFile'].show = false;
+        _state['uploadFile'].transLocale = null;
+        _state['uploadFile'].file = null;
+
+        if(err != null && err.error === true) {
+          console.error(url, err);
+        }
+        if(res != null) {
+          if (res.error) {
+            console.error(url, res.status, res.error.toString());
+            reject(Error(res.error.toString()));
+          } else {
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+      .on('progress', function(e) {
+        _state['uploadFile']['status'] = e.percent;
+      });
+  });
+}
+
+function processDelete(serverResponse) {
+  console.debug('Glossary entry deleted');
+}
+
+function processSaveOrUpdate(serverResponse) {
+  _state['newEntry'].isSaving = false;
+  _state['newEntry'].show = false;
+  _state['newEntry'].isAllowSave = false;
+  _state['newEntry'].pos = '';
+  _state['newEntry'].term = '';
+  _state['newEntry'].description = '';
+
+  console.debug('Glossary entry saved');
+}
+
+function processUpdate(serverResponse) {
+  console.debug('Glossary entry updated');
+}
+
+function initialise() {
+  //load permission here
+  _state['canAddNewEntry'] = canAddNewEntry();
+  _state['canUpdateEntry'] = canUpdateEntry();
+
+  loadLocalesStats()
+    .then(processLocalesStatistic)
+    .then(loadGlossaryByLocale)
+    .then(processGlossaryList)
+    .then(function (newState) {
+      GlossaryStore.emitChange();
+    });
```

Comment:
`newState` is not used. Could also write this in es6 format:

```
.then(() => GlossaryStore.emitChange())
```

or instead of wrapping the call to a function at all, could give a bound version of the function. Not sure if that is really any better though:

```
.then(GlossaryStore.emitChange.bind(GlossaryStore))
```